gems/aws-sdk-s3/lib/aws-sdk-s3/presigner.rb

```diff
@@ -211,16 +211,18 @@ def sign_but_dont_send(
 
           query = http_req.endpoint.query ? http_req.endpoint.query.split('&') : []
           http_req.headers.each do |key, value|
-            next unless key =~ /^x-amz/i
+            next if key == 'user-agent'
 
             if hoist
+              next unless key =~ /^x-amz/i
```

Comment:
is there a case where we would want to hoist range?

Reply:
I'm not convinced we should merge this PR yet, but I made it anyway. I don't think we want to hoist non x-amz headers. I actually couldn't get this to work with a live test (maybe you can give it a try with get_object range and put_object with content-md5).